test/common/router/header_formatter_test.cc

```diff
@@ -167,6 +169,67 @@ TEST_F(RequestInfoHeaderFormatterTest, TestFormatWithUpstreamMetadataVariableMis
   testFormatting(request_info, "UPSTREAM_METADATA([\"namespace\", \"key\"])", "");
 }
 
+TEST_F(RequestInfoHeaderFormatterTest, TestFormatWithPerRequestStateVariable) {
+  Envoy::RequestInfo::FilterStateImpl per_request_state;
+  per_request_state.setData("testing", std::make_unique<StringAccessorImpl>("test_value"));
+  EXPECT_EQ("test_value", per_request_state.getData<StringAccessor>("testing").asString());
+
+  NiceMock<Envoy::RequestInfo::MockRequestInfo> request_info;
+  ON_CALL(request_info, perRequestState()).WillByDefault(ReturnRef(per_request_state));
+  ON_CALL(Const(request_info), perRequestState()).WillByDefault(ReturnRef(per_request_state));
+
+  testFormatting(request_info, "PER_REQUEST_STATE(testing)", "test_value");
+  testFormatting(request_info, "PER_REQUEST_STATE(testing2)", "");
+  EXPECT_EQ("test_value", per_request_state.getData<StringAccessor>("testing").asString());
+}
+
+namespace {
+
+class IntAccessor : public RequestInfo::FilterState::Object {
```

Comment:
I wonder if it is worth it to move this out to test_common, and have it shared here and in the request_info_impl test.

Reply:
I pulled it out into its own file in test/common/request_info (since as shown by the namespace it's a RequestInfo context concept) and referenced that from header_formatter_test.  Let me know what you think.

Reply:
yeah, sounds good to me. Just a small naming change